### PR TITLE
make it work as a separate repository

### PR DESCRIPTION
**Pull request by rogpeppe**
Add license, readme and package doc comment,
and rename exported identifiers to be more appropriate
to a single-purpose package.

### CODE REVIEW

**Review by mhemmings — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -2,24 +2,16 @@ module github.com/juju/mgomonitor
...
-	github.com/juju/clock v0.0.0-20180808021310-bab88fc67299 // indirect
-	github.com/juju/errors v0.0.0-20181012004132-a4583d0a56ea // indirect
-	github.com/juju/loggo v0.0.0-20180524022052-584905176618 // indirect
-	github.com/juju/retry v0.0.0-20180821225755-9058e192b216 // indirect
-	github.com/juju/testing v0.0.0-20180920084828-472a3e8b2073
-	github.com/juju/utils v0.0.0-20180820210520-bf9cc5bdd62d // indirect
-	github.com/juju/version v0.0.0-20180108022336-b64dbd566305 // indirect
 	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
-	github.com/prometheus/client_golang v0.9.0
+	github.com/prometheus/client_golang v0.0.0-20161124155732-575f371f7862
```
Comment (mhilton) [inline go.mod:9]: It'd be nice to stick with what I assume was a newer version.

File: mgo.go [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 import (
 	"github.com/prometheus/client_golang/prometheus"
 	"gopkg.in/mgo.v2"
 )
 
-// MgoStatsCollector is a prometheus.Collector that reports the values
+// Collector is a prometheus.Collector that reports the values
 // provided by mgo.GetStats.
-type MgoStatsCollector struct {
+type Collector struct {
```
Comment (mhilton) [inline mgo.go:17]: This assumes that we never want to support any of the other collectors. That's probably true, but I'm not certain we want to be stuck with that. How about just StatsCollector?

**Review by tasdomas — APPROVED** [review]

**Review by mhilton — APPROVED** [review]
LGTM with a couple of thoughts.

File: go.mod (hunk 1 above)
Comment (rogpeppe) [reply go.mod:9]: I'm using the same version as used by the charmstore. The newer version pulled in some deps I wasn't sure about and I know that prometheus doesn't always keep backward compatibility, so best to be safe I think.

File: mgo.go (hunk 2 above)
Comment (rogpeppe) [reply mgo.go:17]: In the unlikely event that we'll ever need two different MongoDB collectors, we can always choose a new name then. For the time being, I think that "Stats" as a qualifier is pretty meaningless - what would we be collecting from mgo other than statistics?

**Pull request merged**